pkg/fetch/nvd/feed/cpe/v2/v2.go

```diff
@@ -0,0 +1,144 @@
+package v2
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"hash/fnv"
+	"io"
+	"log"
+	"net/http"
+	"path/filepath"
+
+	"github.com/knqyf263/go-cpe/common"
+	"github.com/knqyf263/go-cpe/naming"
+	"github.com/pkg/errors"
+
+	"github.com/MaineK00n/vuls-data-update/pkg/fetch/util"
+	utilhttp "github.com/MaineK00n/vuls-data-update/pkg/fetch/util/http"
+)
+
+const cpeDictionaryURL = "https://nvd.nist.gov/feeds/json/cpe/2.0/nvdcpe-2.0.tar.gz"
+
+type options struct {
+	baseURL string
+	dir     string
+	retry   int
+}
+
+type Option interface {
+	apply(*options)
+}
+
+type baseURLOption string
+
+func (u baseURLOption) apply(opts *options) {
+	opts.baseURL = string(u)
+}
+
+func WithBaseURL(url string) Option {
+	return baseURLOption(url)
+}
+
+type dirOption string
+
+func (d dirOption) apply(opts *options) {
+	opts.dir = string(d)
+}
+
+func WithDir(dir string) Option {
+	return dirOption(dir)
+}
+
+type retryOption int
+
+func (r retryOption) apply(opts *options) {
+	opts.retry = int(r)
+}
+
+func WithRetry(retry int) Option {
+	return retryOption(retry)
+}
+
+func Fetch(opts ...Option) error {
+	options := &options{
+		baseURL: cpeDictionaryURL,
+		dir:     filepath.Join(util.CacheDir(), "fetch", "nvd", "feed", "v2", "cpe"),
```

Comment:
In fetch.go, default dir is `filepath.Join(util.CacheDir(), "fetch", "nvd", "feed", "cpe", "v2")` . Here "cpe" and "v2" are reversed. Nice to use the save path for both?

Reply:
fixed: https://github.com/MaineK00n/vuls-data-update/compare/ed3fdecb66fe73cd91ac7148620c0be2a76209d9..0d9fc64a3955e6a2a65a382d2dff7a52ffc0791b